src/features/resurrection/hooks/useResurrection.ts

```diff
@@ -0,0 +1,75 @@
+import { arrayify } from 'ethers/lib/utils';
+import { useGetSarcophagus } from 'hooks/viewStateFacet';
+import { useGetSarcophagusArchaeologists } from 'hooks/viewStateFacet/useGetSarcophagusArchaeologists';
+import { fetchArweaveFileFallback } from 'lib/utils/arweave';
+import { decrypt } from 'lib/utils/helpers';
+import { useCallback, useEffect, useState } from 'react';
+import { combine } from 'shamirs-secret-sharing-ts';
+
+/**
+ * Hook that handles resurrection of a sarcohpagus
+ * @param sarcoId The sarcohpagus id
+ * @param recipientPrivateKey The recipients private key
+ */
+export function useResurrection(sarcoId: string, recipientPrivateKey: string) {
+  const sarcophagus = useGetSarcophagus(sarcoId);
+  const archaeologists = useGetSarcophagusArchaeologists(sarcoId, sarcophagus?.archaeologists);
+  const [canResurrect, setCanResurrect] = useState(false);
+
+  // Set the canResurrect state based on if the sarcophagus has unencrypted shards
+  useEffect(() => {
+    if (!sarcophagus || !sarcophagus.minShards) return;
+    const shards = archaeologists.filter(a => a.unencryptedShard);
+    setCanResurrect(shards.length >= sarcophagus.minShards);
+  }, [archaeologists, sarcophagus]);
+
+  /**
+   * Resurrects the sarcohpagus using the values passed in to the hook
+   */
+  const resurrect = useCallback(async (): Promise<{ fileName: string; data: string }> => {
+    try {
+      if (!canResurrect) return { fileName: '', data: '' };
+
+      // Get the payload txId from the contract using the sarcoId
+      // The sarcophagus should always have arweave tx ids
+      // The first one is the tx id for the payload, the second is the tx id for the encrypted
+      // shards which is not used here
+      const payloadTxId = sarcophagus.arweaveTxIds[0];
+
+      // In case the sarcophagus has no tx ids. This should never happen but we are checking just in
+      // case.
+      if (!payloadTxId) {
+        throw new Error(`The Arwevae tx id for the payload is missing on sarcophagus ${sarcoId}`);
+      }
```

Comment:
We know there is really no reason for the sarcophagus to not have arweave tx ids on it but the web app doesn't know that. This is just here in case some change is made to the contracts that would cause the arweave tx ids to not be there.